### PR TITLE
Add CBOR and CDDL intro sections

### PR DESCRIPTION
**Pull request by nathanosdev**
I found the section regarding encoding of certificates a bit difficult to follow at first, namely because I was not familiar with CDDL. Google searches will return links regarding "Common Development and Distribution License". Of course if you read the whole spec from start to finish then you will eventually find more information regarding CDDL, but I don't believe this is the way that everyone will read the spec. So I propose adding intro sections near the beginning of the document regarding CDDL and CBOR that can be referenced anywhere these technologies are mentioned.

Other smaller changes in this PR:
- replaced links to obsolete CBOR spec with the current spec
- Fixed one instance of "internet computer" -> "Internet Computer"

### CODE REVIEW

File: spec/index.adoc [hunk 1]
```diff
@@ -344,6 +344,44 @@ where `signing_canister_id` is the id of the signing canister and `reconstruct`
...
+55799({
+  "content": {
+    "request_type": "call",
+    "canister_id": h'ABCD01',
+    "method_name": "say_hello",
+    "arg": h'0061736D01000000'
+  },
+  "sender_sig": h'DEADBEEF',
+  "sender_pubkey": h'B7A3C12DC0C8C748AB07525B701122B88BD78F600C76342D27F25E5F92444CDE'
+})
```
Comment (mraszyk) [inline spec/index.adoc:373]: The other fields (arg, sender_sig, sender_pubkey) are not valid either (I suppose). Do we want this example to be a realistic one?

Comment (nathanosdev) [reply spec/index.adoc:373]: I copied the example from the "CBOR encoding of requests and responses" section. I don't think the example necessarily needs to be realistic, but I also don't think it should be confusing or misleading. The only thing I want to demonstrate here is the before/after of decoding using cbor.me. Maybe it's more sensible to use a basic example from the CBOR spec: https://www.rfc-editor.org/rfc/rfc8949.html#name-examples-of-encoded-cbor-da

I think `82 61 61 a1 61 62 61 63` -> `["a", {"b": "c"}]` could be a nice example, what do you think?

File: spec/index.adoc [hunk 2]
```diff
@@ -344,6 +344,44 @@ where `signing_canister_id` is the id of the signing canister and `reconstruct`
 where `s` is the SHA-256 hash of the `seed` used in the public key and `m` is the SHA-256 hash of the payload.
 --
 
+[#supplementary-technologies]
+== Supplementary Technologies
+
+[#cbor]
+=== CBOR
+
+https://www.rfc-editor.org/rfc/rfc8949[Concise Binary Object Representation(CBOR)] is a data format with a small code footprint, small message size and an extensible interface. CBOR is used extensively throughout the Internet Computer as the primary format for data exchange between components within the system.
```
[the remark this answers is not in the document]
Comment (nathanosdev) [reply spec/index.adoc]: Thanks!

**Review by nomeata — COMMENTED** [review]
Thanks, this is helpful!

File: spec/index.adoc (hunk 1 above)
Comment (mraszyk) [reply spec/index.adoc:373]: I'd prefer an unrelated example in that case (if we do not make the request a proper one).

Comment (nathanosdev) [reply spec/index.adoc]: I've replaced it with the generic example now, thank you

**Review by mraszyk — APPROVED** [review]
Looks good to me. Thank you!

File: spec/index.adoc [hunk 3]
```diff
@@ -344,6 +344,35 @@ where `signing_canister_id` is the id of the signing canister and `reconstruct`
...
+....
+82 61 61 a1 61 62 61 63
+....
+
+Can be converted into the following CBOR diagnostic format:
+....
+["a", {"b": "c"}]
+....
+
+Particular concepts to note from the spec are:
```
Comment (Dfinity-Bjoern) [inline spec/index.adoc:367]:
```suggestion
Particular concepts to note from the spec are:

```

File: spec/index.adoc [hunk 4]
```diff
@@ -344,6 +344,36 @@ where `signing_canister_id` is the id of the signing canister and `reconstruct`
 where `s` is the SHA-256 hash of the `seed` used in the public key and `m` is the SHA-256 hash of the payload.
 --
 
+[#supplementary-technologies]
+== Supplementary Technologies
```
Comment (Dfinity-Bjoern) [inline spec/index.adoc:348]:
```suggestion
=== Supplementary Technologies
```

File: spec/index.adoc [hunk 5]
```diff
@@ -344,6 +344,36 @@ where `signing_canister_id` is the id of the signing canister and `reconstruct`
 where `s` is the SHA-256 hash of the `seed` used in the public key and `m` is the SHA-256 hash of the payload.
 --
 
+[#supplementary-technologies]
+=== Supplementary Technologies
+
+[#cbor]
+=== CBOR
```
Comment (Dfinity-Bjoern) [inline spec/index.adoc:351]:
```suggestion
==== CBOR
```

File: spec/index.adoc [hunk 6]
```diff
@@ -344,6 +344,36 @@ where `signing_canister_id` is the id of the signing canister and `reconstruct`
...
+....
+
+Particular concepts to note from the spec are:
+
+* https://www.rfc-editor.org/rfc/rfc8949#name-specification-of-the-cbor-e[Specification of the CBOR Encoding]
+* https://www.rfc-editor.org/rfc/rfc8949#name-major-types[CBOR Major Types]
+* https://www.rfc-editor.org/rfc/rfc8949#self-describe[CBOR Self-Describe]
+
+[#cddl]
+=== CDDL
```
Comment (Dfinity-Bjoern) [inline spec/index.adoc:374]:
```suggestion
==== CDDL
```

**Review by Dfinity-Bjoern — APPROVED** [review]

**Pull request merged**